### PR TITLE
dm: add doc for dm's ddl compatibility

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

add doc for dm's ddl compatibility.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/9191 https://github.com/pingcap/docs-cn/pull/11584
- Other reference link(s):

### CODE REVIEW

File: dm/dm-ddl-compatible.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: Data Migration DDL Special Handling
+summary:
```
Comment (ran-huang) [inline dm/dm-ddl-compatible.md:3]:
```suggestion
summary: Learn how DM parses and handles DDL statements according to the statement types.
```

File: dm/dm-ddl-compatible.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+---
+title: Data Migration DDL Special Handling
+summary: Learn how DM parses and handles DDL statements according to the statement types.
+---
+
+# Data Migration DDL Special Handling
+
+When DM migrates data, it parses the DDL statement and handles it according to the statement type and the current migration stage.
```
Comment (shichun-0415) [inline dm/dm-ddl-compatible.md:8]:
```suggestion
When TiDB Data Migration (DM) migrates data, it parses the DDL statements and handles them according to the statement type and the current migration stage.
```

File: dm/dm-ddl-compatible.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+|-|-|
+|`^CREATE DATABASE...`|`^CREATE DATABASE...IF NOT EXISTS`|
+|`^CREATE TABLE...`|`^CREATE TABLE..IF NOT EXISTS`|
+|`^DROP DATABASE...`|`^DROP DATABASE...IF EXISTS`|
+|`^DROP TABLE...`|`^DROP TABLE...IF EXISTS`|
+|`^DROP INDEX...`|`^DROP INDEX...IF EXISTS`|
+
+## Shard merge migration tasks
+
+When using the pessimistic mode and optimistic mode for shard merge migration, the behavior of DDL replication is different. For details, refer to [Pessimistic Mode](/dm/feature-shard-merge-pessimistic.md) and [Optimistic Mode](/dm/feature-shard-merge-optimistic.md).
```
Comment (shichun-0415) [inline dm/dm-ddl-compatible.md:139]:
```suggestion
When DM merges and migrates tables  in pessimistic or optimistic mode, the behavior of DDL replication is different frorm that in other scenarios. For details, refer to [Pessimistic Mode](/dm/feature-shard-merge-pessimistic.md) and [Optimistic Mode](/dm/feature-shard-merge-optimistic.md).
```

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

**Pull request merged**